satpy/readers/sgli_l1b.py

```diff
@@ -0,0 +1,266 @@
+
+        full_shape = (self.h5file["Image_data"].attrs["Number_of_lines"],
+                      self.h5file["Image_data"].attrs["Number_of_pixels"])
+
+        tie_lines = np.arange(0, polar_angle.shape[0] * resampling_interval, resampling_interval)
+        tie_cols = np.arange(0, polar_angle.shape[1] * resampling_interval, resampling_interval)
+
+        interpolator = GeoGridInterpolator((tie_lines, tie_cols), azimuthal_angle, polar_angle, method="slinear")
+        new_azi, new_pol = interpolator.interpolate_to_shape(full_shape, chunks="auto")
+        return new_azi, new_pol
+
+    def get_angles(self, key):
+        """Get angles from the file."""
+        if "solar" in key["name"]:
+            azi, zen, attrs = self.get_solar_angles()
+        elif "satellite" in key["name"]:
+            azi, zen, attrs = self.get_sensor_angles()
+        if "azimuth" in key["name"]:
+            dataset = azi
+        else:
+            dataset = zen
+        dataset = xr.DataArray(dataset, attrs=attrs, dims=["y", "x"])
+        return dataset
+
+    def get_solar_angles(self):
+        """Get the solar angles."""
+        azi = self.h5file["Geometry_data/Solar_azimuth"]
+        zen = self.h5file["Geometry_data/Solar_zenith"]
+        attrs = zen.attrs
+        azi = self.scale_array(azi)
+        zen = self.scale_array(zen)
+        return *self.get_full_angles(azi, zen, attrs), attrs
+
+    def get_sensor_angles(self):
+        """Get the solar angles."""
+        azi = self.h5file["Geometry_data/Sensor_azimuth"]
+        zen = self.h5file["Geometry_data/Sensor_zenith"]
+        attrs = zen.attrs
+        azi = self.scale_array(azi)
+        zen = self.scale_array(zen)
+        return *self.get_full_angles(azi, zen, attrs), attrs
+
+    def scale_array(self, array):
+        """Scale an array with its attributes `Slope` and `Offset` if available."""
+        try:
+            return array * array.attrs["Slope"] + array.attrs["Offset"]
+        except KeyError:
+            return array
+
+    def get_full_angles(self, azi, zen, attrs):
+        """Interpolate angle arrays."""
+        resampling_interval = attrs["Resampling_interval"]
+        if resampling_interval != 1:
+            zen = zen[:] - 90
+            new_azi, new_zen = self.interpolate_spherical(azi, zen, resampling_interval)
+            return new_azi, new_zen + 90
+        return azi, zen
+
+
+class H5Array(BackendArray):
+    """An Hdf5-based array."""
+
+    def __init__(self, array):
+        """Initialize the array."""
+        self.shape = array.shape
+        self.dtype = array.dtype
+        self.array = array
+
+    def __getitem__(self, key):
+        """Get a slice of the array."""
+        return indexing.explicit_indexing_adapter(
+            key, self.shape, indexing.IndexingSupport.BASIC, self._getitem
+        )
+
+    def _getitem(self, key):
+        return self.array[key]
+
+
+class SGLIBackend(BackendEntrypoint):
+    """The SGLI backend."""
```

Comment:
Since most of us are unfamiliar with xarray backends, could you add to the docstring here and for the H5Array about what they are and how they work? Are these even used in Satpy?